### PR TITLE
Added tests for rows and cols

### PR DESCRIPTION
**Pull request by johnedwardobrien**
Added tests. Updated the 'Gemfile.lock' on my local repo so that I could get rspec working, don't know if that particular change really needs to be merged with your master, will let you be the judge of that.

### CODE REVIEW

File: spec/grid_spec.rb [hunk 1]
```diff
@@ -36,4 +36,30 @@
 
     expect(result).to eq(false)
   end
+
+    it "does not place queen if queen is already in row" do
```
Comment (ChrisBr) [inline spec/grid_spec.rb:40]: [nitpick] Indentation is wrong here

**Review by ChrisBr — APPROVED** [review]
Congrats on your first PR 👏

**Comment by johnedwardobrien** [comment]
Okay thanks I'll pay attention to that a bit more.

**Comment by johnedwardobrien** [comment]
I want to write more tests to get more familiar with rspec and best
practices. If queen-puzzle had all of the tests that it needed would that
mean that it had tests testing all of its functionality like the tests you
wrote and then I mimicked? Or would tests for every function, method, and
construction element as it appears be considered satisfactory when all
written?

**Comment by ChrisBr** [comment]
> I want to write more tests to get more familiar with rspec and best
practices. If queen-puzzle had all of the tests that it needed would that
mean that it had tests testing all of its functionality like the tests you
wrote and then I mimicked? Or would tests for every function, method, and
construction element as it appears be considered satisfactory when all
written?

It's usually not necessary (and often impossible) to test every single method and class. What you want to aim for is to test your public interface / API in our case that would probably be the `BruteForceSolver` or if we go down one layer the `Grid`. I wouldn't test for instance `Move` or `Diagonal` because we can easily test this behaviour through `Grid` and we also couple our tests to the implementation. What I mean with coupling is that if we test `Grid` (and through `Grid` the `Move` and `Diagonal` behaviour) it gives us a lot of freedom to change our implementation without breaking our tests (which would require additional work to fix them again). Testing is a complicated topic and there are a lot of different opinions (and this is mine) but I think it's safe to say that testing all classes and methods is usually not required.

This is also a very good talk about what to test: [Sandy Metz - Magic Tricks of Testing](https://www.youtube.com/watch?v=URSWYvyc42M)

**Comment by johnedwardobrien** [comment]
Coupling the tests to implementation and API makes sense, thanks.

**Pull request merged**